utils/cq/daos_pylint.py

```diff
@@ -496,8 +527,38 @@ def main():
         rc_tmp.flush()
         args.rcfile = rc_tmp.name
 
+    if args.diff:
+        # There are to be two different ways of driving this, for a commit-hook where we only
+        # check files changed and in GitHub actions where we check the entire tree and are more
+        # strict for files which have changed.
+        if args.git:
+            all_files = get_git_files()
+        else:
+            all_files = FileTypeList()
+        regions = None
+        for line in sys.stdin.readlines():
+            if line.startswith('diff --git a/'):
+                parts = line.split(' ')
+                file = parts[3][2:-1]
+                regions = OutPutRegion()
+                all_files.add_regions(file, regions)
+                if not args.git:
+                    all_files.add(file)
+                continue
+            if line.startswith('@@ '):
+                parts = line.split(' ')
+                (post_start, post_len) = parts[2][1:].split(',')
+                regions.add_region(int(post_start), int(post_len))
```

Comment:
There might be an off-by-one error here although given that we're using diff -U10 it probably doesn't matter in practice.